### ISSUE TITLE
Indent-eating is not contained in partials

### ISSUE THREAD

**Issue by KitsuneRal** [kainjow/Mustache#27]
As of Mustache 4.0b2 (but not before), partials eat spaces to the left of them in certain cases. Specifically, if `passPartial` is defined as `{{#existingSection}}{{/existingSection}}` and `partial` is `{{#existingSection}}Text{{/existingSection}}` then
```
No indent
    {{#nonExistentSection}}{{/nonExistentSection}}Normal behaviour
    {{#existingSection}}{{/existingSection}}Ditto
    {{>passPartial}}Indentation gets eaten
    {{>partial}}Indentation preserved
No indent
```
brings
```
No indent
    Normal behaviour
    Ditto
Indentation gets eaten
    Indentation preserved
No indent
```

**Comment by kainjow** [comment]
Sorry for the year+ delay. I finally reproduced the bug and attempted a fix with fb872ae721b73c9eec52ef55b3a3022862a76728. If you're able to verify still that'd be great. I'm going to do some more testing though to see what the behavior is with multi-nested partials.

**Comment by KitsuneRal** [comment]
Sure, will try, hopefully some time throughout the week.
